gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/test/java/io/gravitee/gateway/handlers/api/context/ContentTemplateVariableProviderTest.java

```diff
@@ -0,0 +1,281 @@
+ */
+package io.gravitee.gateway.handlers.api.context;
+
+import static io.gravitee.gateway.handlers.api.context.ContentTemplateVariableProvider.*;
+import static io.gravitee.gateway.jupiter.api.context.RequestExecutionContext.TEMPLATE_ATTRIBUTE_REQUEST;
+import static io.gravitee.gateway.jupiter.api.context.RequestExecutionContext.TEMPLATE_ATTRIBUTE_RESPONSE;
+import static org.mockito.Mockito.*;
+
+import io.gravitee.el.TemplateContext;
+import io.gravitee.el.TemplateEngine;
+import io.gravitee.gateway.api.buffer.Buffer;
+import io.gravitee.gateway.jupiter.api.context.Request;
+import io.gravitee.gateway.jupiter.api.context.RequestExecutionContext;
+import io.gravitee.gateway.jupiter.api.context.Response;
+import io.gravitee.gateway.jupiter.api.el.EvaluableRequest;
+import io.gravitee.gateway.jupiter.api.el.EvaluableResponse;
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.observers.TestObserver;
+import java.util.Collections;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * @author Jeoffrey HAEYAERT (jeoffrey.haeyaert at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@ExtendWith(MockitoExtension.class)
+class ContentTemplateVariableProviderTest {
+
+    protected static final String REQUEST_CONTENT = "request content";
+    protected static final String REQUEST_JSON_CONTENT = "{\"request\": \"content\"}";
+    protected static final String REQUEST_XML_CONTENT = "<request>content</request>";
+
+    protected static final String RESPONSE_CONTENT = "response content";
+    protected static final String RESPONSE_JSON_CONTENT = "{\"response\": \"content\"}";
+    protected static final String RESPONSE_XML_CONTENT = "<response>content</response>";
+
+    @Mock
+    private RequestExecutionContext ctx;
+
+    @Mock
+    private TemplateEngine templateEngine;
+
+    @Mock
+    private TemplateContext templateContext;
+
+    @Mock
+    private Request request;
+
+    @Mock
+    private Response response;
+
+    @Mock
+    private EvaluableRequest evaluableRequest;
+
+    @Mock
+    private EvaluableResponse evaluableResponse;
+
+    private final ContentTemplateVariableProvider cut = new ContentTemplateVariableProvider();
+
+    @BeforeEach
+    void init() {
+        lenient().when(ctx.request()).thenReturn(request);
+        lenient().when(ctx.response()).thenReturn(response);
+
+        when(ctx.getTemplateEngine()).thenReturn(templateEngine);
+        when(templateEngine.getTemplateContext()).thenReturn(templateContext);
+        when(templateContext.lookupVariable(TEMPLATE_ATTRIBUTE_REQUEST)).thenReturn(evaluableRequest);
+        when(templateContext.lookupVariable(TEMPLATE_ATTRIBUTE_RESPONSE)).thenReturn(evaluableResponse);
+    }
+
+    @Test
```

Comment:
maybe you could add some `DisplayName` for the tests ?

Reply:
We should see if we adopt this good practice more globally.
For now I tried to be verbose enough so don't see a big issue if we don't add `DisplayName` for now but I'll keep it in mind for the next tests I'll wrote 👍